### PR TITLE
Issue 3415 recaptcha DRAFT

### PR DESCRIPTION
**Pull request by cesy**
#3415 says we need more security - trying adding recaptcha. Not yet fully tested, not ready to merge, but putting it public for early discussion.
Using tips from https://blog.dennisokeeffe.com/blog/2022-03-08-part-6-adding-recaptcha-to-prevent-fraud-and-abuse

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -738,7 +740,7 @@ DEPENDENCIES
   xmlrpc
 
 RUBY VERSION
-    ruby 3.1.4p223
+   ruby 3.1.4p223
```
[the remark this answers is not in the document]
Comment (cesy) [reply Gemfile.lock]: This seems to happen every time we regenerate Gemfile.lock - any bright ideas, or shall I just edit manually again?

[the remark this answers is not in the document]
Comment (CloCkWeRX) [reply Gemfile.lock]: It will eventually be fixed in bundler at some point. Did hunt down the discussion

[the remark this answers is not in the document]
Comment (CloCkWeRX) [reply Gemfile.lock]: (edit manually for now)

**Comment by cesy** [comment]
Next steps:
* Fix the trailing space thing
* Since we can't access our old shared Google account, either set up a new one or figure out how to use my personal one.
* Test it in a codespace or other dev env
* Change PR from draft to ready for review once that works, then deploy to staging, set up config on prod, deploy to prod, test
At the moment we're running over the free limit on emails every month due to the spam.

**Comment by cesy** [comment]
I've set up a site on Google's reCAPTCHA v2 using my personal email, and I can add co-owners - let me know if you want adding.

**Comment by cesy** [comment]
Testing in codespace, got a step further, but it's complaining about a missing site key which I think means env variables are not loading properly even when I've copied env-example to application.yml and restarted server. Next thing to try is setting the relevant env variable manually in bash and checking it's saved and rails can see it.

**Comment by CloCkWeRX** [comment]
This seems pretty close, from the code point of view.

File: app/views/devise/registrations/new.html.haml [hunk 2]
```diff
@@ -3,9 +3,10 @@
     %h1 Join #{ENV['GROWSTUFF_SITE_NAME']}
     .card-body
       %p Sign up for a #{ENV['GROWSTUFF_SITE_NAME']} account to track your vegetable garden and connect with other local growers.
+      %p If you have accessibility issues with the captcha, please contact us via the links in the footer and we will help.
 
     = bootstrap_form_for(resource, as: resource_name, url: registration_path(resource_name),
-        html: { class: "text-center border border-light p-5" }) do |f|
+        html: { class: "text-center border border-light p-5", data: { turbo: false} }) do |f|
```
Comment (CloCkWeRX) [inline app/views/devise/registrations/new.html.haml]:
```suggestion
        html: { class: "text-center border border-light p-5", data: { turbo: false } }) do |f|
```

**Comment by CloCkWeRX** [comment]
From bin/setup
```
  unless File.exist?('.env')
    FileUtils.cp 'env-example', '.env'
  end
```

image

**Comment by CloCkWeRX** [comment]
Going to replace with https://github.com/Growstuff/growstuff/pull/3586 so everyone can commit to it.

**Comment by CloCkWeRX** [comment]
@cesy - did you want to configure staging with the credentials?
https://dashboard.heroku.com/apps/growstuff-staging/settings

**Pull request closed without merging**